Fix array compilation

NadavPeled1998 opened the pull request:
There is a bug with array compilation in the js2leo files.

e.g, If there is an array of private fields property inside a record, the array will be compiled in the js2leo files as:

`array_property:  js2leo.privateField(js2leo.array(array_property, js2leo.field))`

The `privateField` function expects to get a string, so when you try to run a test with the program that has this record, you will immediately get this error: Type 'string' is not assignable to type 'string[]'.

I fixed this issue, and after this PR, the array will be compiled to:

`array_property:  js2leo.array(js2leo.array(array_property, js2leo.field), js2leo.privateField)`

Please review and approve this PR ASAP. Any project that uses records with array properties can't use doko-js because of that.

NadavPeled1998 commented:
There is a critical bug with testing functions that there are inputs/outputs of multi-dimensional arrays and arrays with custom types.
I added a commit to fix it.
 If there are any questions or comments on this PR, please let me know 🙏

anilpdl commented:
@NadavPeled1998 Thanks for the update.
Can you provide the use-case for this, and some sample program that we could replicate and test on the fix?

NadavPeled1998 commented:
@anilpdl I added a test for a function with inputs, outputs, and records with field array, custom type array, multi-dimensional field array, and multi-dimensional custom type array.
While creating the tests, I found an issue in records with a multi-dimensional array(fixed).
I also fixed existing bugs in tests, so it's now possible to run `npm run test` multiple times without errors.

anilpdl commented:
@NadavPeled1998 We making some additional updates to support the functionalities on all supported execution modes (dry run, execute mode and deployed version). This might be delayed further until we are ready for release.
Thanks for the updates and apologies for any inconveniences.
